pkg/handlers/generic/lifecycle/addons/helmaddon.go

```diff
@@ -143,36 +145,26 @@ func (a *helmAddonApplier) Apply(
 		targetCluster = applyOpts.targetCluster
 	}
 
-	if chartProxy == nil {
-		chartProxy = &caaphv1.HelmChartProxy{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: caaphv1.GroupVersion.String(),
-				Kind:       "HelmChartProxy",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: targetCluster.Namespace,
-				Labels: map[string]string{
-					clusterv1.ClusterNameLabel: cluster.Name,
-					ClusterNamespaceLabel:      cluster.Namespace,
-					// Label values have a maximum length of 63 characters so hash the release name to
-					// ensure it fits within the limit.
-					HelmReleaseNameHashLabel: hashReleaseName(a.config.defaultHelmReleaseName),
-				},
-				GenerateName: fmt.Sprintf("%s-", a.config.defaultHelmReleaseName),
+	chartProxy := &caaphv1.HelmChartProxy{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: caaphv1.GroupVersion.String(),
+			Kind:       "HelmChartProxy",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: targetCluster.Namespace,
+			Name:      fmt.Sprintf("%s-%s", a.config.defaultHelmReleaseName, clusterUUID),
```

Comment:
Great change on how to generate the name! 

nit: Do you think its still valuable to have the cluster name label here to make it selectable for debugging?

Reply:
Without the workload cluster namespace I don't think so, you can't guarantee to uniquely identify it from the cluster name if there are two clusters with same name in different namespaces for example.